### PR TITLE
Add connection to Neo4j and basic network endpoints

### PR DESCRIPTION
**Pull request by chaoran-chen**
This PR integrates the network into the Icebreaker application. The corresponding frontend PR is https://github.com/icebreaker-science/frontend/pull/20.

The data are stored in a Neo4j database. `database/docker-compose.yml` was extended so that it should be easy to start a Neo4j instance on the local machine (@reviewer, please try it out).

There is no integration with `WikiPage` yet. There are also no tests since it is an early prototype and probably the subject to many changes.

### CODE REVIEW

File: src/main/java/science/icebreaker/config/ApplicationNeo4jConfigurationProperties.java [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+    private String host;
+
+    private int port;
+
+    private String username;
+
+    private String password;
+
+
+    public String getHost() {
```
Comment (bieniusa) [inline src/main/java/science/icebreaker/config/ApplicationNeo4jConfigurationProperties.java:20]: Adding public getters *and* setters for private fields is pointless if they only directly reset the object attribute. 
However, I am not going to fight over this, if all of the code follows this convention.

File: src/main/java/science/icebreaker/network/NetworkController.java [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+            @ApiParam(value = "Name of a node", example = "raman spectroscopy")
+            @RequestParam("node") String node
+    ) {
+        return service.getEgoGraph(node);
+    }
+
+
+    @GetMapping("/graph/shortest_path")
+    @ApiOperation("Returns the edges of a graph containing the selected nodes and the shortest paths between all pairs")
+    public List<Edge> getShortestGraphGraph(
```
Comment (bieniusa) [inline src/main/java/science/icebreaker/network/NetworkController.java]: Should this be "getShortestPathGraph"?

File: src/main/java/science/icebreaker/network/NetworkController.java [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+
+    @GetMapping("")
+    @ApiOperation("Returns the number of nodes - just for fun. This endpoint can change any time and should not be " +
+            "actually used.")
+    public int getNumberOfNodes() {
+        return service.countNodes();
+    }
+
+
+    @GetMapping("/node")
```
Comment (bieniusa) [inline src/main/java/science/icebreaker/network/NetworkController.java:34]: If it returns more than one, I would name the path "nodes".

File: src/main/java/science/icebreaker/network/NetworkService.java [hunk 4]
```diff
@@ -0,0 +1,105 @@
...
+import java.util.stream.Collectors;
+
+import static org.neo4j.driver.Values.parameters;
+
+
+@Service
+public class NetworkService implements AutoCloseable {
+
+    /**
+     * TODO Find a better solution to limit the size of the results.
```
Comment (bieniusa) [inline src/main/java/science/icebreaker/network/NetworkService.java:17]: Is there a related issue?

**Review by bieniusa — CHANGES_REQUESTED** [review]

File: src/main/java/science/icebreaker/network/NetworkService.java (hunk 4 above)
Comment (chaoran-chen) [reply src/main/java/science/icebreaker/network/NetworkService.java:17]: I have added a task on the Notion page for the network.

File: src/main/java/science/icebreaker/network/NetworkController.java (hunk 3 above)
Comment (chaoran-chen) [reply src/main/java/science/icebreaker/network/NetworkController.java:34]: The names of the other endpoints such as `/wiki` and `/device_availability` are also in the singular form even though they return multiple values, so I would like to keep it the same way.

File: src/main/java/science/icebreaker/network/NetworkController.java (hunk 2 above)
Comment (chaoran-chen) [reply src/main/java/science/icebreaker/network/NetworkController.java]: yes

File: src/main/java/science/icebreaker/config/ApplicationNeo4jConfigurationProperties.java (hunk 1 above)
Comment (chaoran-chen) [reply src/main/java/science/icebreaker/config/ApplicationNeo4jConfigurationProperties.java:20]: yeah.. but that's java.. I think that the Spring framework even expects getters and setters.

**Review by bieniusa — APPROVED** [review]

**Pull request merged**